### ISSUE TITLE
[Feature Request] Long press on back button brings you back to the main timeline

### ISSUE THREAD

**Issue by bonsaiheldin** [tuskyapp/Tusky#947]
Tusky remembers the last viewed pages and lets me go back page by page using the back button, which is good. But often i want to just go back to the main stream immediately and then i have to click on the back button multiple times, often leading to minimizing the app because i was too fast and tapped too often.

I've come up with a, as i think, fairly intuitive solution to this: A long press on the back button, which makes Tusky forget all viewed pages and brings you back to the main timeline immediately.

* * * *
- Tusky Version: 4.0
- Android Version: 6.0.1
- Android Device: Xiaomi Redmi 3S
- Mastodon instance (if applicable): chaos.social

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by MasterGroosha** [comment]
I hope you're mentioning on-screen "back" button in top right corner?

I personally think that messing with system controls (navigation bar or hardware keys) is not a good idea.

**Comment by bonsaiheldin** [comment]
> I hope you're mentioning on-screen "back" button in top right corner?
> 
> I personally think that messing with system controls (navigation bar or hardware keys) is not a good idea.

Yes, i'm talking about the button of the app itself, not about the phone's button. I will include that info in my issue. Thanks for asking!

**Comment by charlag** [comment]
Maybe side menu should be accessible everywhere :thinking:

**Comment by ichier** [comment]
in twidere the in app-button always went back to the main timeline you were coming from after clicking trough some threads and links.
and i already might remember it wrong, but i believe the android back button went step by step.

no matter if i am remembering correctly, this would be my suggestion: 

- the in-app back buttons brings you back to the timeline, you were coming from
- while the systems back button brings you back step by step

this might be irritating at first, but once you got it, pretty helpful, giving you both options right at hand: the "more intuitive" one lays closer to where your fingers are, the (otherwise redundant) in-app button provides the bonus big jump feature, serving all camps this way.
